platform/mbed_retarget.cpp

```diff
@@ -880,6 +880,9 @@ extern "C" WEAK void __iar_file_Mtxinit(__iar_Rmtx *mutex) {}
 extern "C" WEAK void __iar_file_Mtxdst(__iar_Rmtx *mutex) {}
 extern "C" WEAK void __iar_file_Mtxlock(__iar_Rmtx *mutex) {}
 extern "C" WEAK void __iar_file_Mtxunlock(__iar_Rmtx *mutex) {}
+#if (__IAR_SYSTEMS_ICC__ >= 8)
+extern "C" WEAK void *__aeabi_read_tp (void) {}
```

Comment:
Thread Local Storage requirement has changed in this version of IAR, and requires RTOS to allocate memory for secondary threads as part of TLS support. read_tp() is to point to the TLS section. 

We do not have support of TLS in current RTOS, hence weak function is added to assist build. Actual support will be added in future with changes in RTOS.